tensorboard/backend/security_validator.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Validates responses and their security features"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import re
+
+from werkzeug import wrappers
+from werkzeug.datastructures import Headers
+from werkzeug import http
+
+from tensorboard.util import tb_logging
+
+Policy = collections.namedtuple("Poicy", ["name", "rules"])
```

Comment:
Overall if we're writing something general-purpose here (for parsing any CSP that a plugin emits), it would be nice to use the terminology from the spec: https://www.w3.org/TR/CSP/#framework-directives

So in particular the policy would actually be the entire value of the CSP header.  I think this object would actually be what the spec calls a Directive, which has a directive name (fine as is) and then a directive value.  The directive value is always a set of strings and it looks like this specifically a "source list" for almost all directives, so we could perhaps just call it `source_list`?